### PR TITLE
Flux architecture

### PR DESCRIPTION
**Pull request by LukaszDygon**
I've implemented popularly used Flux architecture for storage and retrieval of entities

### CODE REVIEW

**Comment by Joe-Edwards** [comment]
Is there a particular reason for this change? Do you think the direct approach is more difficult?

Bear in mind this is a training project for Java, so we want to keep the frontend code as simple as possible.

Slightly related - is there a reason to drop all the semicolons? It's not a complete consensus, but the vast majority of code uses them.

Sorry, I know you've been left to get on with this by yourself - it looks like a neat approach for a proper project, I'm just concerned about whether it's suitable for apprentice training.

**Comment by LukaszDygon** [comment]
@Joe-Edwards 
The flux is something I've only learned about about recently and I do agree that it is making the system a little more complex, but actually accessing the data you want is much easier than calling the API directly from the component every time. It's also more easily extensible.  I decided to implement it at the point I did because I found handling all the functionality becoming messy and increasingly asking myself where in the structure should different functionality go, and this approach makes it more systematic.
I can revert this to the previous direct approach if that's still not justifying the added architecture complexity for this project.

I've removed semi-colons because I find it makes code easier to read and work with. Pretty much the only active contributor to the frontend code at this point, with Jack working mostly with the backend, so this was personal preference. I can put them back in, if that's preferable for the course. I am aware that most programmers still prefer having the semi-colons, but as far as I know, they became obsolete with ES6.

In summary, I am not attached to any of the decisions, so I am happy to abandon them both if it's for the benefit of the project's goal.

**Comment by Joe-Edwards** [comment]
> In summary, I am not attached to any of the decisions, so I am happy to abandon them both if it's for the benefit of the project's goal.

Sorry, I think flux looks really interesting but yeah, I'd basically like to abandon them :(

> I've removed semi-colons because I find it makes code easier to read and work with.

They're mandatory in Java, so I think removing them in JavaScript won't save very much mental effort.

The bigger problem with removing semicolons is that it's fine 95% of the time, but just occasionally it catches you out:

```
foo()
[1, 2, 3, 4].forEach(bar)
```

This will _not_ insert a semicolon - it'll parse as indexing into the result of `foo()` which is almost certainly not what you want!

Perhaps the best thing would be to add eslint (or similar), which can automatically insert missing semis.
